main.py

```diff
@@ -192,19 +228,73 @@ async def does_code_match(code, member):
             continue
         id = parsed[0]
         parsed_code = parsed[1]
+        email = parsed[2]
         if id == str(member.id) and parsed_code == code:
+            await send_verified_log(str(email), member)
             return True
     return False
 
+
+# get the uoft email associated with the user
+async def get_verified_email(user):
+    global verified_channel
+    messages = await verified_channel.history(limit=20000).flatten()
+    for message in messages:
+        parsed = message.content.split(", ")
+        email = parsed[0]
+        id = parsed[2]
+        if id == str(user.id):
+            break
+
+    await send_banned_log(email, user)
+    print("user: " + str(user.name) + " was banned from the server, the email: " + str(
+        email) + " was added to ban list")
+
+    #TODO user not found case
+
+
+# check if email has been banned or already in use
+async def is_free_email(user, email):
```

Comment:
This implementation of "already in use" might break if a user needs a secondary code to be sent?